File: backend/app/public/schemas/location_service_days.py
from uuid import UUID
from datetime import date

from pydantic import BaseModel


class ListLocationServiceDaysFilterRequest(BaseModel):
    date_from: date | None = None
    date_to: date | None = None
    is_open: bool | None = None


class LocationServiceDayTableResponse(BaseModel):
    id: UUID
    location_service_id: UUID
    service_type_id: int

    service_name: str
    service_date: date

    is_open: bool
    capacity: int

File: frontend/src/lib/api/public/types/location-service-days.ts
import { z } from 'zod';

import { dateSchema } from '$lib/types/dates';

// Location Service Day Filters

export const listLocationServiceDaysFilterRequestSchema = z.object({
	date_from: z.iso.date().nullable().optional(),
	date_to: z.iso.date().nullable().optional(),
	is_open: z.boolean().nullable().optional()
});

export type ListLocationServiceDaysFilterRequest =
	z.infer<typeof listLocationServiceDaysFilterRequestSchema>;


// Location Service Day Table Response

export const locationServiceDayTableResponseSchema = z.object({
	id: z.uuid(),
	location_service_id: z.uuid(),
	service_type_id: z.number(),

	service_name: z.string(),
	service_date: dateSchema,

	is_open: z.boolean(),
	capacity: z.number()
});

export type LocationServiceDayTableResponse =
	z.infer<typeof locationServiceDayTableResponseSchema>;